soaktest/soakclient_transaction/soakclient_transaction.go

```diff
@@ -0,0 +1,552 @@
+					return transactionCommitNum, err
+				}
+				rewindConsumerPosition(c)
+				return transactionCommitNum, nil
+			} else {
+				fmt.Errorf("SendOffsetsToTransaction() failed: %s\n", err)
+				return transactionCommitNum, err
+			}
+		} else {
+			break
+		}
+	}
+
+	if transactionCommitNum%100 != 0 {
+		// If CommitTransaction succeed, transactionCommitNum + 1 and return
+		// If CommitTransaction failed with err.(kafka.Error).IsRetriable(),
+		// sleep SLEEP_TIME then retry
+		// If CommitTransaction failed with err.(kafka.Error).TxnRequiresAbort(),
+		// AbortTransaction and return (transactionCommitNum, err)
+		// If failed with other errors, return transactionCommitNum and the
+		// fatal error
+		for i := 0; i < RETRY_NUM; i++ {
+			err = p.CommitTransaction(nil)
+			if err != nil {
+				if i == 0 {
+					soaktest.DatadogIncrement(
+						PRODUCER_TRANSACTION_COMMIT_FAILED,
+						1)
+				}
+				fmt.Printf("CommitTransaction() failed: %s\n", err)
+				if err.(kafka.Error).IsRetriable() {
+					if i == RETRY_NUM-1 {
+						fmt.Printf("CommitTransaction() failed with max "+
+							"retry %d times: %s\n", RETRY_NUM, err)
+						return transactionCommitNum, err
+					}
+					time.Sleep(SLEEP_TIME)
+					continue
+				} else if err.(kafka.Error).TxnRequiresAbort() {
+					err = p.AbortTransaction(nil)
+					if err != nil {
+						fmt.Printf("AbortTransaction() failed: %s\n", err)
+						return transactionCommitNum, err
+					}
+					err = rewindConsumerPosition(c)
+					if err != nil {
+						fmt.Printf("rewindConsumerPosition() failed: %s\n", err)
+					}
+					return transactionCommitNum, nil
+				} else {
+					fmt.Errorf("CommitTransaction() failed: %s\n", err)
+					return transactionCommitNum, err
+				}
+			} else {
+				soaktest.DatadogIncrement(
+					PRODUCER_TRANSACTION_COMMIT_SUCCEED,
+					1)
+				transactionCommitNum += 1
+				return transactionCommitNum, nil
+			}
+		}
+	} else {
+		// If AbortTransaction succeed, transactionCommitNum = 1 and return
+		// If AbortTransaction failed with err.(kafka.Error).IsRetriable(),
+		// sleep SLEEP_TIME then retry
+		// If failed with other errors, return transactionCommitNum and the
+		// fatal error
+		for i := 0; i < RETRY_NUM; i++ {
+			err = p.AbortTransaction(nil)
+			if err != nil {
+				if i == 0 {
+					soaktest.DatadogIncrement(
+						PRODUCER_TRANSACTION_ABORT_FAILED,
+						1)
+				}
+				fmt.Printf("AbortTransaction() failed: %s\n", err)
+				if err.(kafka.Error).IsRetriable() {
+					if i == RETRY_NUM-1 {
+						fmt.Printf("AbortTransaction() failed with max "+
+							"retry %d times: %s\n", RETRY_NUM, err)
+						return transactionCommitNum, err
+					}
+					time.Sleep(SLEEP_TIME)
+					continue
+				} else {
+					fmt.Printf("AbortTransaction() failed: %s\n", err)
+					return transactionCommitNum, err
+				}
+			} else {
+				soaktest.DatadogIncrement(
+					PRODUCER_TRANSACTION_ABORT_SUCCEED,
+					1)
+				transactionCommitNum = 1
```

Comment:
Is this supposed to be an increment or reset?

Reply:
Hi Magnus, This is reset